### ISSUE TITLE
Clean up Paper 1

### ISSUE THREAD

**Issue by Glavin001** [Glavin001/Wearables-Research#13]
Paper 1: https://github.com/Glavin001/Wearables-Research/tree/master/papers/muse-users-activity-identification

- [ ] Remove figure 10
- [ ] Remove figure 11
- [ ] Combine Precision columns in Table 2 into subcolumns
  - Precision (2 sub columns), Entire dataset, ten-fold cross-validation
- [ ] Explanation of Variable names (for Variable Importance)
  - B1.Beta.1 = Bin 1, Beta wave, Location 1
- [ ] Add more minsplit tests for Decision Tree
- [ ] Fix SVM and Random Forest accuracy results

**Comment by mtriff** [comment]
Completed first three items.

I have started editing and cleaning up sections to prepare for the paper submissions, I will be committing a separate `.tex` file for these changes due to the difference in scope between the papers and your thesis. I am only working on the first four sections at this point, so any updates to the remaining sections can be copied over once they are completed. Thanks!

**Comment by Glavin001** [comment]
Excellent! I'm currently working on paper 2 ( #9 ) with Genetic Algorithms. Once I am home in a few days and have a better Internet connection I will be reviewing the API documentation and source code for the third-party packages with the hope that I can fix the code that generates the models and accuracy tables. I'll let you know if I make any progress on the front. Thanks!

**Comment by Glavin001** [comment]
Paper submission: http://www.ucalgary.ca/icci_cc/iccicc-16/paper-submission

**Comment by Glavin001** [comment]
Note the page limit is 8 pages!

Remaining To Do:
- [ ] Abstract (Pawan)
- [ ] Experimental Design (Glavin)
  - Exclude Genetic Algorithms
  - Include:
    - Decision Tree
    - Support Vector Machine
    - Neural Network
    - Random Forest 
- [ ] Results and Discussions
- [ ] Organize Graphs and Figures (Glavin)
  - [x] Create new `data_files.txt` that includes 50 sessions from Toby
    - We now have 173 sessions total from Bruce (23), Jerry (50), Matt (50), Toby (50)
  - [ ] Update all plots
    - [x] Table for entire dataset stats
  - [ ] Decision tree example could be entire page, landscape
  - [ ] Follow Matt's template
- [ ] Conclusion 

Clean up:  
- [ ] Use IEEE formatting  (Matt)
- [ ] Review of Wearables - from other paper (Matt)
- [ ] Study Data - from other paper (Matt)
- [ ] Knowledge Representation - from other paper (Matt)

---

Configuring our classification algorithms:

- Neural Network
  - More than 100 is too many
- Decision Tree
  - Minsplit from 1 to 20
- SVM
  - People like most: Gaussian and RDF

**Comment by Glavin001** [comment]
I fixed it!

Here is the updated table with the correct precision for ten-fold cross validation:

![image](https://cloud.githubusercontent.com/assets/1885333/13029489/8167380c-d264-11e5-8db7-859a1dd74799.png)
![image](https://cloud.githubusercontent.com/assets/1885333/13029490/8356887a-d264-11e5-816a-9b499beede71.png)

**Comment by pawanLingras** [comment]
Good job. This makes complete sense and can be properly explained in the
paper.
If you guys update the paper over next couple of days, I can add my bit
while in transit.

Thanks.
Pawan.

**Comment by Glavin001** [comment]
I apologize for not getting back sooner.
@mtriff could you and I get on a call tomorrow to clean things up?

I need some help with:
- [ ] making the confusion matrix tables using your template (may no longer be applicable, see below)
- [ ] cleaning up the decision tree plots
- [ ] which parameters I should include in the `Experimental Design` section (we listed algorithm parameters in WWCI16 paper)

I will keep working away until we can connect. I think I am close to finishing. It does not yet feel polished yet.

### Decision Tree

I am surprised that the builtin R and even third-party packages for plotting the decision trees looks so poor. Is there a *standard* way to plot decision trees that I am somehow missing?

Version 1:
![image](https://cloud.githubusercontent.com/assets/1885333/13131928/4e7653ba-d5c5-11e5-8db5-c093f16f68e6.png)

Version 2:
![image](https://cloud.githubusercontent.com/assets/1885333/13131934/55d3e302-d5c5-11e5-8163-5ed896398a68.png)

Version 3:
![image](https://cloud.githubusercontent.com/assets/1885333/13131938/5bbc04fc-d5c5-11e5-83a2-290477f8449a.png)

**Update**: Version 4:
![image](https://cloud.githubusercontent.com/assets/1885333/13132419/d05d0b0a-d5c8-11e5-86fa-4d14b43ad1c6.png)


And then of course we could photoshop it manually. 
I need some help with what our ideal is for how it should look like and then I can form it.

### Confusion Matrix tables

Is this the template you mentioned?

![image](https://cloud.githubusercontent.com/assets/1885333/13131749/1440fea8-d5c4-11e5-827d-0ce855d1ef71.png)

It turns out that even the best Decision Tree created for Activity & Person+Activity was just a root node. So all we will need is the Person confusion matrix. Here is the confusion matrix for entire dataset and 10-fold CV confusion matrices (values are percentages over the repeats):
![image](https://cloud.githubusercontent.com/assets/1885333/13132179/1471dac0-d5c7-11e5-8efb-409f1fe49048.png)

To show that Activity decision tree and confusion matrix is rather dull:
![image](https://cloud.githubusercontent.com/assets/1885333/13132190/2abf4bd2-d5c7-11e5-885d-9938ff0312cc.png)


### Tables

Also the results tables are too wide:
![image](https://cloud.githubusercontent.com/assets/1885333/13131662/777d0c74-d5c3-11e5-886d-5ef2b4fcbf57.png)

- Do you have suggestions for informative names for the classifiers that will be shorter?
- Are there too many? I can delete a few of the rows.

**Comment by Glavin001** [comment]
I finally was able to get the tables to work from your template. It turns out I was pesky missing a column in the alignment field and it was causing my tables to prevent the paper from building.

I've pushed my work on `Experimental Design` and `Results and Discussions`.
I think we should:
- [ ] discuss the Decision Tree plot(s) situation
- [ ] review the `Experimental Design` that I have completed
- [ ] finish choosing which Confusion matrices to discuss in the `Results and Discussions` section (I have added many for us to choose from)
- [ ] review the rest of `Results and Discussions` section that I have completed above

**Comment by mtriff** [comment]
Hi Glavin,

Since the packages you have tried aren't working, I would say try using
something like https://www.draw.io to create what you need manually.

Glad you were able to get the table working, I will review those sections.
The tables that are too wide should go on their own page. Once we have
settled on the tables, I can do that.

Pawan may want to weigh in on which confusion matrices to use in the
discussion.

Thank you,
Matt

**Comment by pawanLingras** [comment]
We will use only the SVM and NN (1000 iterations) confusion matrices. I am
assuming you can get those using the table function. If you can not easily
get them, let us not worry about it.

**Comment by Glavin001** [comment]
> Since the packages you have tried aren't working, I would say try using
something like https://www.draw.io to create what you need manually.

I will manually create the Decision Tree based on the *data* from this one:
![image](https://cloud.githubusercontent.com/assets/1885333/13131938/5bbc04fc-d5c5-11e5-83a2-290477f8449a.png)

And *look* a little more like this:
![image](https://cloud.githubusercontent.com/assets/1885333/13152494/e3a1cc8c-d645-11e5-9d4a-743307d16211.png)

To confirm, what would you prefer to see?
- [ ] Boxes or not boxes around Node labels?
- [ ] Node labels are anonymous (0,1,2,3) or names (Bruce, Jerry, Matt, Toby)?
- [ ] Percentages / Probabilities on Nodes labels?
- [ ] Decision rules (i.e. `B1.Beta.1 < 35`) is below the Node (ex #1 above) or split up into leaves (ex #2 above)?
- Anything else that would be useful to see when reading a paper?

> The tables that are too wide should go on their own page. Once we have
settled on the tables, I can do that.

Thanks, I appreciate it.

> We will use only the SVM and NN (1000 iterations) confusion matrices. 

Excellent, I think I had originally added Random Forest, SVM, and NN anyway.
I will polish those confusion matrices for Person, Activity, and Person+Activity and finish off the `Results and Discussions` section about them.

---

After you look over `Experimental Design` and `Results and Discussions` let me know if there is anything I should add.

Would we like to include tables on the `Recall` performance as well as the `Precision`? Maybe `Recall` is something for later, I could include in my thesis paper and not for this paper.

**Comment by Glavin001** [comment]
Here's my current drawing of the Person Decision Tree:

![wearables research - decision tree - person](https://cloud.githubusercontent.com/assets/1885333/13154963/d40fb49e-d651-11e5-8774-bfc2b07a0da3.png)

Link to edit: https://drive.google.com/file/d/0B5-qlJU7RXiNd0p0emRnUHRZV3c/view?usp=sharing

**Comment by Glavin001** [comment]
I have added the confusion matrices and updated the decision tree image.

Here is a PDF of the current paper: [wiechertTriffLingrasICCI_CC16.pdf](https://github.com/Glavin001/Wearables-Research/files/136790/wiechertTriffLingrasICCI_CC16.pdf)

**Comment by Glavin001** [comment]
Here is the latest clean decision tree:
![decision-tree-person-clean](https://cloud.githubusercontent.com/assets/1885333/13188040/5e32d556-d725-11e5-8b2d-c5a19dc6a071.png)

---

I have added a seed (`set.seed(100)`), such that any future runs will generate the same results and our tables and figures are all in sync, however now that I have the updated results they have changed slightly from the conclusion.

![image](https://cloud.githubusercontent.com/assets/1885333/13188088/a79367ec-d725-11e5-84d0-6ce03f26a571.png)
![image](https://cloud.githubusercontent.com/assets/1885333/13188093/af0cc518-d725-11e5-99ae-5f0d0c8e5fbb.png)
![image](https://cloud.githubusercontent.com/assets/1885333/13188097/ba61a8fc-d725-11e5-9253-1fbbdeac402b.png)

Most notably, SVM and Neural Networks went from ~97% to 95.39% and 94.46% respectively, which brings them very close to Random Forest at 93.85%. Should we continue to only include SVM and Neural Networks or should we also include Random Forest confusion matrices?

**Comment by Glavin001** [comment]
@mtriff: I have pushed some more paragraphs discussing the results. I left in the `%Glavin` markers so you can see where I have made my changes below them.

---

I noticed when analyzing the confusion matrix tables that it appears Toby (individual 3) has an extra 5 records added, making 23+50+50+55=178 records in total. I noticed because with 100\% precision from using the entire dataset there were 55 positive predictions for class 0. 
I was able to confirm this is the case. It appears that Rstudio was using a cached, older selection of records -- where I had selected 55 instead of 50 records for Toby. The results are still valid, except that Toby had 55 records instead of the 50 as we wrote.

I could either change the sentences that reference the number of records for each individual or re-run the scripts with Toby at 50 records and update the tables where applicable. Let me know what you think is the best plan of action. I am OK with either plan, as it was my mistake for not double checking the data was the expected 173 records instead of 178.

I'll make any more changes you need tomorrow so we can complete this paper. Thank you.

**Comment by mtriff** [comment]
Good work with the additions, they look good. My only remaining question
has to do with the first comment I made. In the preceding sentence you
mention that the method "allowed us to study the importance of different
attributes". Which attributes were determined to be more important? Which
attributes were determined to be less important? Why?

I am on the fence about running scripts again with 50 of Toby's records or
just leaving it as is with 55 records.

Pawan, do you have a preference?

Thanks,
Matt

**Comment by Glavin001** [comment]
> In the preceding sentence you
mention that the method "allowed us to study the importance of different
attributes". Which attributes were determined to be more important? Which
attributes were determined to be less important? Why?

I think I started answer that question and then I moved what I wrote below into `Results and Discussions` section. Would this be more appropriate in the `Results and Discussion` section or the `Experimental Design`, where I mentioned that the method "allowed us to study the importance of different
attributes"?

Also, I only have tables for the top 10 significant variables. Should I add a table for the least 10 significant variables as well so we can discuss them?

**Comment by Glavin001** [comment]
I'm at SMU helping out a friend study for a midterm so since I am tied up I am letting my laptop run over all of the data again and re-generating the tables. Once it is done, I will swap in the new, correct results (23,50,50,50=173 records) and update the discussions and conclusion with the appropriate values.
Sorry for the inconvenience and delays. I hope to have my sections completed by the end of tonight. Thank you.

**Comment by Glavin001** [comment]
While I'm making changes, would it be good to also remove some of Bruce's records? From 23 to 15 records (3 for each of the 5 activities), so 15+50+50+50=165 records. I noticed a bias towards `game` and when I investigated further I saw that Bruce had significantly more records.

```
data/Bruce/game/1/combined.csv
data/Bruce/game/10/combined.csv * Remove
data/Bruce/game/2/combined.csv
data/Bruce/game/3/combined.csv
data/Bruce/game/4/combined.csv * Remove
data/Bruce/game/5/combined.csv * Remove
data/Bruce/game/6/combined.csv * Remove
data/Bruce/game/7/combined.csv * Remove
data/Bruce/game/8/combined.csv * Remove
data/Bruce/game/9/combined.csv * Remove

data/Bruce/music/1/combined.csv
data/Bruce/music/2/combined.csv
data/Bruce/music/3/combined.csv
data/Bruce/music/4/combined.csv * Remove

data/Bruce/none/1/combined.csv
data/Bruce/none/2/combined.csv
data/Bruce/none/3/combined.csv

data/Bruce/reading/1/combined.csv
data/Bruce/reading/2/combined.csv
data/Bruce/reading/3/combined.csv

data/Bruce/video/1/combined.csv
data/Bruce/video/2/combined.csv
data/Bruce/video/3/combined.csv
```

Running the training and validation is not taking too long so I can do it again tomorrow morning and finish everything off for the early afternoon.

Let me know what you would like me to do and I can finish everything off. Thank you.

**Comment by pawanLingras** [comment]
Glavin: We do not want to make changes at this stage. Evening out games may make Bruce's records even smaller.

**Comment by pawanLingras** [comment]
Same thing applies to Toby. No changes to the experiment. Just change the numbers to the correct ones in the paper.

**Comment by Glavin001** [comment]
@pawanLingras : I have corrected the numbers.

@mtriff : I have added most discussion about the significant independent variables. I am concerned about adding too much discussion to the `Experimental Design` section as you requested. I opted for a shorted discussion of the significant variables within the `Experimental Design` section and more details available within the `Results and Discussions`. 

PDF of latest paper: [wiechertTriffLingrasICCI_CC16.pdf](https://github.com/Glavin001/Wearables-Research/files/145038/wiechertTriffLingrasICCI_CC16.pdf)

**Comment by pawanLingras** [comment]
I went through the paper, with a particular focus on the results and discussion section. It reads well. I do not like to nitpick. It is your writing and I think it is good quality. From this point on, I am inclined to just say that it is ready, when you think it is ready. We have a little more space, may be bringing back the image of the muse band may not be a bad idea.
